architecture/2024-11-authz-feature-spec.md

```diff
@@ -0,0 +1,392 @@
+* Granting and revoking permission to users to perform create, read, update, delete actions on resources and resource groups including applications, user-defined resource types, environments, and recipe registrations
+* Granting and revoking permission to users to associate resources between resource groups
+
+### Non-goals (out of scope)
+
+This feature specification only addresses authorization. A separate feature specification is under development for authenticating users.
+
+## User profiles and challenges
+
+**IAM administrator** – This person is not a Radius user, but rather someone who administers the organization's identity and access management or identity provider system. The IAM administrator models the organization's structure into a set of user groups which will be leveraged by external systems such as Radius. These groups are stored as user to group mapping in an identity provider such as Microsoft Entra ID, Okta, and Ping Identity. 
+
+**Radius administrator** – The Radius administrator needs to model their organization's structure into a set of resource groups within Radius and grant access to these groups. The Radius administrator must have awareness of the user groups stored in the identity provider in order to associate a set of users with a resource group. Therefore, in some organizations, the Radius administrator may have some access to the identity provider system.
+
+**Developers** – Developers need to use the resource types available to them to model their applications and deploy to environments which they have access to.
+
+**Environment administrators** – An environment administrator needs to create and maintain environments in Radius for developers to deploy their applications to. The environment administrator must ensure the physical or cloud environment is prepared to have applications deployed to it and that the Radius environment is configured correctly. In some organizations, the environment administrator will determine who has permission to deploy applications to which environment.
+
+**Resource type administrators** – A resource type administrator needs to maintain the resource types which are available in the Radius system and manage who can use those resource types within their applications.
+
+**Recipe administrators** – A recipe administrator needs to manage the set of recipes registered to an environment. They must author, test, and maintain infrastructure as code modules such as Terraform or Bicep to implement the recipe. In many organizations, a recipe administrator and an environment administrator will be the same person. However, in other organizations, administration of recipes may be delegated to specialty teams for certain resource types; e.g., DBAs may maintain recipes for database resource types.
+
+## Scenarios
+
+The following scenarios will describe a final setup similar to the image below.
+
+![](2024-11-authz-feature-spec/authz.png)
+
```

Comment:
A summary of the user, roles, the permissions we are thinking and what it allows the user to do would be helpful